wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,191 @@
+    template_name = "wagtailadmin/reports/page_types_usage.html"
+    title = _("Page types usage")
+    header_icon = "doc-empty-inverse"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.user_model = get_user_model()
+        self.page_models = [model.__name__.lower() for model in get_page_models()]
+        self.i18n_enabled = getattr(settings, "WAGTAIL_I18N_ENABLED", False)
+
+    def get_filterset_class(self):
+        if self.i18n_enabled:
+            self.filterset_class = PageTypesUsageReportFilterSet
+        return super().get_filterset_class()
+
+    def user_id_to_python(self, user_id):
+        return self.user_model._meta.pk.to_python(user_id)
+
+    def add_last_edited_name_to_page_type(self, username_mapping, page_type):
+        if page_type.last_edited_by:
+            # Giving the last_published_by annotation an explicit output_field type
+            # causes an issue with prefetch_workflow_states when the field is a
+            # ConvertedValueField. If the last user to publish the page has been
+            # deleted, we will render their user id in the template, so we call
+            # to_python on the value so that what's rendered matches the developer's
+            # expectation in the case of complex primary key types (e.g. UUIDField).
+            try:
+                user_id_value = self.user_id_to_python(page_type.last_edited_by)
+            except ValidationError:
+                user_id_value = page_type.last_edited_by
+
+            last_edited_by_user = username_mapping.get(
+                user_id_value, get_deleted_user_display_name(user_id=user_id_value)
+            )
+            page_type.last_edited_by_user = last_edited_by_user
+
+    def add_last_edited_page_to_page_type(self, pages_mapping, page_type):
+        if page_type.last_edited_page:
+            last_edited_page = pages_mapping.get(page_type.last_edited_page, None)
+            page_type.last_edited_page = last_edited_page
+
+    def decorate_paginated_queryset(self, page_types):
+        page_ids = set(page_types.values_list("last_edited_page", flat=True))
+        pages_mapping = {page.pk: page for page in Page.objects.filter(pk__in=page_ids)}
+
+        user_ids = set(page_types.values_list("last_edited_by", flat=True))
+        username_mapping = {
+            user.pk: user.get_username()
+            for user in self.user_model.objects.filter(pk__in=user_ids)
+        }
+
+        for page_type in page_types:
+            self.add_last_edited_page_to_page_type(pages_mapping, page_type)
+            self.add_last_edited_name_to_page_type(username_mapping, page_type)
+        return page_types
+
+    def get_queryset(self):
+        queryset = ContentType.objects.filter(model__in=self.page_models)
+        self.queryset = queryset
+
+        # 'edited at' info is handled at the filter level, since ContentType itself does
+        # not have a locale to filter on
+        if self.i18n_enabled:
```

Comment:
Is this guarding conditional necessary? I would argue it's the role of `filter_queryset()` to avoid applying filters unnecessarily.

Reply:
That's true, nice catch. Before, since there was only the locale filter, the filterset would be disabled if `i18n` was disabled. Since now we have both a locale and a site, I can modify this.

Reply:
Done!